Sources/GRPCNIOTransportCore/Client/Connection/LoadBalancers/RoundRobinLoadBalancer.swift

```diff
@@ -750,12 +754,16 @@ extension ConnectivityState {
       return .idle
     }
 
-    // Otherwise, if all subchannels are in state TRANSIENT_FAILURE, the channel's state
-    //   is TRANSIENT_FAILURE.
-    if states.allSatisfy({ $0 == .transientFailure }) {
-      return .transientFailure
+    // Otherwise, if all subchannels are in state TRANSIENT_FAILURE, the channel's state is TRANSIENT_FAILURE.
+    // Pick one of the errors to surface, as we can't surface all of them.
+    var cause: RPCError!
+    for state in states {
+      guard case .transientFailure(let error) = state else {
+        return .shutdown
+      }
+      cause = error
     }
```

Comment:
Can we exhaustively switch over the state here? I think it helps make the code more obviously correct:

```swift
var cause: RPCError?
for state in states {
  switch state {
  case .transientFailure(let error):
    // ...
  case .shutdown:
    return .shutdown
  case .idle, .connecting, .ready:
    fatalError("...")  // Unreachable as all are covered above.
  }
}
```

We also know that if `cause` is `nil` after this point it can only be so because  `states` is empty. IMO this makes it clearer that the returning `.shutdown` when there's no state is a fallback rather than behaviour we were implementing from some spec (as we have done for the other cases).